core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/common/AbstractUserStoreManager.java

```diff
@@ -16171,4 +16174,19 @@ protected void handleGetNonExistentUser(String userName, String profileName) thr
         handleGetUserClaimValuesFailure(errorCode, errorMessage, userName, null, profileName);
         throw new UserStoreException(errorCode + " - " + errorMessage);
     }
+
+    /**
+     * The password validity timeout value is set by server configuration value from carbon.xml file.
+     * If value is not present the default value of 24 hours is returned.
+     * @return password validity timeout in hours.
+     */
+    protected int getDefaultPasswordValidityPeriodInHours() throws UserStoreException {
```

Comment:
calling this method each time is costly. Let's have a class level static variable and call this method once and resolve it. Then we could reuse it

Reply:
changed the function to use a static Integer and read value from carbon.xml only if the value is previously not read.